### PR TITLE
doc: improve Catalina discoverability for search engines

### PR DESCRIPTION
**Pull request by mmarchini**
By showing the error in the guide, search engines should be able to find
this document when users search for specific lines from the error.

##### Checklist

- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: macOS_Catalina.md [hunk 1]
```diff
@@ -2,7 +2,23 @@
...
+
+No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.
+
+gyp: No Xcode or CLT version detected!
+gyp ERR! configure error
+gyp ERR! stack Error: `gyp` failed with exit code: 1
+gyp ERR! stack     at ChildProcess.onCpExit (/Users/wtodd/.nvm/versions/node/v13.10.1/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
+gyp ERR! stack     at ChildProcess.emit (events.js:316:20)
+gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
+gyp ERR! System Darwin 19.4.0
```
Comment (cclauss) [inline macOS_Catalina.md:20]:
```suggestion
gyp: No Xcode or CLT version detected!
```
My sense is that this error message should be sufficient.  The other lines might might change ("/Users/wtodd", etc.).

Comment (cclauss) [reply macOS_Catalina.md:20]: We should also mark this class of issues with the ` gyp: No Xcode or CLT version detected! ` label.

Comment (mmarchini) [reply macOS_Catalina.md:20]: Accepted your suggestion, but, what about using this instead:

```
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
```

I can imagine folks searching for the first three lines as well (not everyone will know which lines are the most relevant to search for). Agreed that the last ones are platform dependent (I thought I changed those before committing :stuck_out_tongue:)

**Comment by rvagg** [comment]
963f2a7

**Pull request closed without merging**